WordPress/gutenberg
# Add a 'View Site' link in the site editor

## What?
Add a link from the site editor to view the site.

## Why?
Because it is currently not within easy reach.

## How?
Added a new `MenuItem` to `PreviewOptions` in edit-site/components/header.

As a part of this update I've changed 'Preview' to 'View' for a couple of reasons:

1. There's no way to _pre_view site editor changes yet.
2. The device view options aren't strictly previews, they are just view modes for the canvas.

## Testing Instructions
1. Open the Site Editor
2. Click 'View' in the top toolbar
3. Click 'View Site'
4. Ensure the site homepage is opened in a new tab

## Screenshots or screencast 


https://user-images.githubusercontent.com/846565/178288203-cf071aa1-85aa-431b-8680-11eeee14941c.mp4
